pkg/webhook/validatingwebhook/validate.go

```diff
@@ -0,0 +1,119 @@
+package validatingwebhook
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/pkg/errors"
+	rbac "k8s.io/api/rbac/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "k8s.io/api/admission/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var (
+	runtimeScheme = runtime.NewScheme()
+	codecs        = serializer.NewCodecFactory(runtimeScheme)
+	deserializer  = codecs.UniversalDeserializer()
+
+	log = logf.Log.WithName("users_rolebindings_validating_webhook")
+)
+
+func HandleValidate(w http.ResponseWriter, r *http.Request) {
+	var respBody []byte
+	body, err := ioutil.ReadAll(r.Body)
+	defer func() {
+		if err := r.Body.Close(); err != nil {
+			log.Error(err, "unable to close the body")
+		}
+	}()
+	if err != nil {
+		log.Error(err, "unable to read the body of the request")
+		w.WriteHeader(http.StatusInternalServerError)
+		respBody = []byte(fmt.Sprintf("unable to read the body of the request: %s", err))
+	} else {
+		// validate the request
+		respBody = validate(body)
+		w.WriteHeader(http.StatusOK)
+	}
+	if _, err := w.Write(respBody); err != nil {
+		log.Error(err, "unable to write response")
+	}
+}
+
+func validate(body []byte) []byte {
+	admReview := v1.AdmissionReview{}
+	if _, _, err := deserializer.Decode(body, nil, &admReview); err != nil {
+		log.Error(err, "unable to deserialize the admission review object", "body", body)
+		admReview.Response = responseWithError(admReview.Request.UID, err)
+	}
+	// let's unmarshal the object to be sure that it's a rolebinding
+	var rb *rbac.RoleBinding
+	if err := json.Unmarshal(admReview.Request.Object.Raw, &rb); err != nil {
+		log.Error(err, "unable unmarshal rolebinding json object", "AdmissionReview", admReview)
+		admReview.Response = responseWithError(admReview.Request.UID, errors.Wrapf(err, "unable unmarshal rolebinding json object - raw request object: %v", admReview.Request.Object.Raw))
+	}
+	requestingUser := admReview.Request.UserInfo
+	subjects := rb.Subjects
+	allServiceAccountsSubject := rbac.Subject{
+		Kind:     "Group",
+		Name:     "system:serviceaccounts",
+		APIGroup: "rbac.authorization.k8s.io",
+	}
+	allUsersSubject := rbac.Subject{
+		Kind:     "Group",
+		Name:     "system:authenticated",
+		APIGroup: "rbac.authorization.k8s.io",
+	}
+
+	//check the requesting user is not a system user
+	if !strings.Contains(requestingUser.Username, "system:") {
```

Comment:
@MatousJobanek isn't it going to be problematic with App Studio when multiple users are given permissions on Space's namespaces?

Reply:
also, you could probably use somethings like `strings.HasPrefix` instead of `strings.Contains`

Reply:
Just checking if the username starts with `system:` won't work. Not because of multiple users per space but because SA account names start with `system:`. But our goal is not to block all users but service accounts. Our goal is to prevent Sandbox users from creating such role bindings. Other regular users should be fine (dedicated admin users, crt users, other potential sre users, etc). Such users don't have the `system:` prefix.
So, it seems that we have to get the corresponding OpenShift user object and check the labels to determine if the user was created / is managed by Sandbox. If so then reject.
Any other ideas how we could do it differently?
Sandbox users still would be able to create role bindings if they use a service account but IMO that's fine. Our main goal is to prevent accidental role binding creation.

Reply:
@xcoulon using HasPrefix now